### PR TITLE
WIP: Tech 4/configuration dsl

### PR DESCRIPTION
**Pull request by mhenrixon**
This is just a suggestion on how we could wrap the semantic_logger in a way that enables us to control the defaults for the logging and the metrics for all projects without having to duplicate the configuration in each project that uses the gem.

It is just a proposal, all feedback deeply appreciated. What we probably want to avoid is to spread `SemanticLogger` out throughout our codebase as to avoid future problems with refactoring.

### CODE REVIEW

**Comment by reevoo-samuel** [comment]
*If you're new, read the [guidelines]
(https://github.com/reevoo/guidelines/blob/master/pull_requests.md).*

Good code is **tested**, **designed for change**, **easy to understand**
and **able to cope when things go wrong**.

Good pull requests are **descriptive**, **small** and **short-lived**.

**For the Author**
- [ ] Description explains what you have done and why.
- [ ] Tests have been written (or you have explained why you don't need
tests).
- [ ] Documentation/comments have been added/updated.

**For the Reviewer**
- [ ] The code is well-designed and able to cope with change.
- [ ] The code is easy to understand.
- [ ] Tests cover happy and unhappy paths in the code.
- [ ] Tests are green on CI (or pulled/tested on your machine).

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,9 @@
-source 'https://rubygems.org'
+source "https://rubygems.org"
 
 # Specify your gem's dependencies in sapience.gemspec
 gemspec
+
+gem "pry-nav"
+gem "sentry-raven"
+gem "statsd-ruby"
+gem "semantic_logger", github: "mhenrixon/semantic_logger"
```
Comment (robinbortlik) [inline Gemfile]: Just curious, why did you fork the `semantic_logger` gem, are you going to make some custom changes in it?

Comment (mhenrixon) [inline Gemfile]: For starters they didn't have an appender for Sentry (Raven). I cloned their repo to create a pull request for that but I can't really use their repo until they merge it.

Comment (mhenrixon) [inline Gemfile]: https://github.com/rocketjob/semantic_logger/pull/31

Comment (robinbortlik) [inline Gemfile]: 👍

**Comment by robinbortlik** [comment]
LGTM 👍

File: lib/sapience.rb [hunk 2]
```diff
@@ -1,5 +1,42 @@
 require "sapience/version"
+require "sapience/sapience"
 
+# @formatter:off
 module Sapience
-  # Your code goes here...
+  autoload :AnsiColors,         "sapience/ansi_colors"
```
Comment (errm) [inline lib/sapience.rb]: Is there a good reason for using autoload, it is on the whole not thread safe, and it is much easier to make stuff work reliably in puma if we just require everything up front.

File: lib/sapience.rb [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+    autoload :Raw,              "sapience/formatters/raw"
+    autoload :Syslog,           "sapience/formatters/syslog"
+  end
+
+end
+# @formatter:on
+
+# Close and flush all appenders at exit, waiting for outstanding messages on the queue
+# to be written first
+at_exit do
```
Comment (errm) [inline lib/sapience.rb]: If we did this as a signal trap then I think we would not bump heads with rspec etc

**Comment by reevoo-samuel** [comment]
Er, hello?

This Pull Request is too damn old! Merge or close this, sucka.

**Comment by reevoo-samuel** [comment]
Er, hello?

This Pull Request is too damn old! Merge or close this, sucka.

**Pull request merged**